packages/react/dropdown-menu/src/DropdownMenu.tsx

```diff
@@ -266,38 +265,18 @@ const DropdownMenuRootContent = React.forwardRef<
       {...contentProps}
       ref={forwardedRef}
       portalled={portalled}
-      onCloseAutoFocus={(event) => {
-        props.onCloseAutoFocus?.(event);
-
-        if (!event.defaultPrevented) {
-          if (!hasInteractedOutsideRef.current) context.triggerRef.current?.focus();
-          // Always prevent auto focus because we either focus manually or want user agent focus
-          event.preventDefault();
-        }
-
+      onCloseAutoFocus={composeEventHandlers(props.onCloseAutoFocus, (event) => {
+        if (!hasInteractedOutsideRef.current) context.triggerRef.current?.focus();
         hasInteractedOutsideRef.current = false;
-      }}
-      onInteractOutside={(event) => {
-        props.onInteractOutside?.(event);
-
-        if (!event.defaultPrevented) {
-          const originalEvent = event.detail.originalEvent as PointerEvent;
-          const ctrlLeftClick = originalEvent.button === 0 && originalEvent.ctrlKey === true;
-          const isRightClick = originalEvent.button === 2 || ctrlLeftClick;
-
-          if (!context.modal || isRightClick) hasInteractedOutsideRef.current = true;
-        }
-
-        // Prevent dismissing when clicking the trigger.
-        // As the trigger is already setup to close, without doing so would
-        // cause it to close and immediately open.
-        //
-        // We use `onInteractOutside` as some browsers also
-        // focus on pointer down, creating the same issue.
-        const target = event.target as HTMLElement;
-        const targetIsTrigger = context.triggerRef.current?.contains(target);
-        if (targetIsTrigger) event.preventDefault();
```

Comment:
I've added the `onOpenToggle` handlers back in and everything seems to be working fine without this.

Can anyone recall which browser this was an issue in? I've tested both modal and non-modal in Firefox, Chrome, Brave, Safari, iOS Safari and Edge.

Reply:
I can't recall anything specific, but it seems to work well.